src/messaging/ExchangeDelegate.h

```diff
@@ -61,13 +61,13 @@ class ExchangeContext;
  *        on the exchange, then the exchange remains with you, and it's your responsibility to either send a message on it,
```

Comment:
I think item 4 needs to change to "No message send attempts happened" or something, because of the behavior change in SendMessage where it sends the will-send flag...  It's very unclear whether "no messages were sent on that exchange" refers to SendMessage calls or actual messages going out.  Need to figure out how to document the behavior here clearly.

This is a must-fix.

Reply:
So, I've decided to revert the setting of the WillSendMessage flag early on in SendMessage. That hopefully makes this text relevant/correct again?

Reply:
Yes, that looks good.